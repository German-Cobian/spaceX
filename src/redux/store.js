import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import missionsReducer from './missionsReducer';

const reducer = combineReducers({
  missionsReducer,
  // additional reducers could be added here
});

const store = createStore(
  reducer,
  applyMiddleware(logger, thunk),
);

export default store;
