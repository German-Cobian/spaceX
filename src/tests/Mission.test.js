import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import PrintMission from '../components/Mission';
import store from '../redux/store';

test('matches RocketsProfile component snapshot', () => {
  const missionsData = [{
    // eslint-disable-next-line max-len
    description: "Eutelsat S.A. is a European satellite operator. Providing coverage over the entire European continent, the Middle East, Africa, Asia and the Americas, it is the world's third largest satellite operator in terms of revenues.",
    id: 'F7709F2',
    name: 'Eutelsat',
    reserved: false,
  }];

  const tree = renderer.create(
    <Provider store={store}>
      <PrintMission list={missionsData} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
