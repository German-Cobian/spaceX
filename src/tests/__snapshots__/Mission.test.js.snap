// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches RocketsProfile component snapshot 1`] = `
<tr>
  <th>
    Eutelsat
  </th>
  <td>
    Eutelsat S.A. is a European satellite operator. Providing coverage over the entire European continent, the Middle East, Africa, Asia and the Americas, it is the world's third largest satellite operator in terms of revenues.
  </td>
  <td
    className="text-center"
    id="container1"
  >
    <span
      className="badge bg-secondary"
    >
      NOT A MEMBER
    </span>
  </td>
  <td
    className="text-center"
    id="container"
  >
    <button
      className="btn btn-outline-secondary"
      disabled={false}
      id="F7709F2"
      onClick={[Function]}
      type="button"
    >
      Join Mission
    </button>
  </td>
</tr>
`;
